contract/test/proveEligibility.test.js

```diff
@@ -0,0 +1,423 @@
+    nextUpdate,
+    notifier,
+    async checkNotifier() {
+      const { value: balance, updateCount } = await nextUpdate;
+      console.log('checking notiifer:::', { updateCount, balance });
+      nextUpdate = await E(notifier).getUpdateSince(updateCount);
+      console.log('nextUpdate:::', { nextUpdate });
+    },
+  };
+};
+const trace = label => value => {
+  console.log(label, '::::', value);
+  return value;
+};
+test('airdrop purses', async t => {
+  const { initialSupply, bonusSupply } = makeMemesSupply(
+    1_000_000n,
+    1_500_000n,
+  );
+
+  const [primaryPurse, secondaryPurse] = [
+    memeIssuer.makeEmptyPurse().deposit(memeMint.mintPayment(initialSupply)),
+    memeIssuer.makeEmptyPurse().deposit(memeMint.mintPayment(bonusSupply)),
+  ]
+    .map(PurseHolder)
+    .map(Id);
+
+  const getProp = prop => obj => obj[prop];
+  const getValue = getProp('value');
+  const getPurse = getProp('purse');
+
+  const getBalance = compose(getPurse, getValue);
+
+  t.deepEqual([primaryPurse, secondaryPurse].map(getBalance), [
+    initialSupply,
+    bonusSupply,
+  ]);
+});
+
+const handleValidateProof =
+  (tree = TEST_TREE_DATA.tree, hash = TEST_TREE_DATA.rootHash) =>
+  (proof = TEST_TREE_DATA.proofs[0], nodeValue = TEST_TREE_DATA.leaves[0]) =>
+    tree.verify(proof, nodeValue, hash);
+
+test('merkle tree verification', t => {
+  const verifyAgainstTestTree = handleValidateProof();
+
+  t.deepEqual(
+    verifyAgainstTestTree(),
+    true,
+    'verifyAgainstTestTree function given default arguments, should return true',
+  );
+
+  t.deepEqual(
+    verifyAgainstTestTree(TEST_TREE_DATA.proofs[1], accounts[1].pubkey.key),
+    true,
+    'handleValidateProof function given a proof and its corresponding account should return true',
+  );
+
+  t.deepEqual(
+    verifyAgainstTestTree(TEST_TREE_DATA.proofs[1], 'notarealpubkey'),
+    false,
+    'handleValidateProof function given proof and a pubkey value that does not exist in the tree should return false',
+  );
+});
+
+test('airdrop claim :: eligible participant', async t => {
+  await makeTestContext(t);
+
+  const { publicFacet, timer, testTreeRemotable } = await t.context;
+
+  const validateFn = await E(testTreeRemotable).getVerificationFn();
+
+  t.deepEqual(
+    validateFn(TEST_TREE_DATA.proofs[0], accounts[0].pubkey.key),
+    true,
+    'TreeRemotable should expose function that properly verifies a proof against a Merkle tree',
+  );
+  await E(timer).advanceTo(2719838800n);
+
+  await simulateClaim(
+    t,
+    await E(publicFacet).makeClaimInvitation(),
+    memes(1000n),
+    {
+      proof: TEST_TREE_DATA.proofs[0],
+      pubkey: accounts[0].pubkey.key,
+      address: accounts[0].address,
+    },
+  );
+
+  await simulateClaim(
+    t,
+    await E(publicFacet).makeClaimInvitation(),
+    memes(1000n),
+    {
+      proof: TEST_TREE_DATA.proofs[2],
+      pubkey: accounts[2].pubkey.key,
+      address: accounts[2].address,
+    },
+  );
+
+  await simulateClaim(
```

Comment:
@zmanian this is where the tests end at. hoping to make the `simulateClaim` more generic to the point where it could handled different cases, in this case failure. otherwise, will define offer handler code for non-success claim attempts within the body of the test.